### PR TITLE
feat: Focus approval request UI when a request is queued

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

Currently when a request is queued, it does not cause the existing confirmation window to be focused. This is because queued requests are withheld from the `ApprovalController` which is responsible for showing batched/scrubbable confirmations to the user. Since the `ApprovalController` never actually receives queued requests until they are ready to be shown to the user, it becomes the responsibility of the `QueuedRequestController` to determine when the confirmation window must be focused because a new confirmation has been enqueued.

This PR adds a new callback/hook to the `QueuedRequestController`, enabling it to trigger the notification window receiving focus.

## References

Fixes: https://github.com/MetaMask/metamask-extension/issues/25397

### `@metamask/queued-request-controller`

- **BREAKING**: `QueuedRequestController` constructor params now requires a `showApprovalRequest` hook that is called when the approval request UI should be opened/focused as the result of a request with confirmation being enqueued.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/queued-request-controller/src/QueuedRequestController.ts [hunk 1]
```diff
@@ -83,6 +83,7 @@ export type QueuedRequestControllerOptions = {
     request: QueuedRequestMiddlewareJsonRpcRequest,
   ) => boolean;
   clearPendingConfirmations: () => void;
+  showApprovalRequest: () => void;
```
Comment (jiexi) [inline packages/queued-request-controller/src/QueuedRequestController.ts:86]: should this be optional?..

Comment (adonesky1) [reply packages/queued-request-controller/src/QueuedRequestController.ts:86]: probably not? this controller will only be used on the extension AFAIK and I believe we will always want this hook passed?

**Review by adonesky1 — APPROVED** [review]
LGTM!

File: packages/queued-request-controller/src/QueuedRequestController.ts (hunk 1 above)
Comment (jiexi) [reply packages/queued-request-controller/src/QueuedRequestController.ts:86]: it's optional in the sense that this isn't required for the QueuedRequestController to perform it's core function

Comment (adonesky1) [reply packages/queued-request-controller/src/QueuedRequestController.ts:86]: sure thats definitely true. I guess I think of the type as safeguard for accidental removal... though maybe that's not really true

**Pull request merged**